resolwe_bio/processes/reads_processing/bbduk_beta.py

```diff
@@ -0,0 +1,1380 @@
+            f"hammingdistance={inputs.processing.hamming_distance}",
+            f"qhdist={inputs.processing.query_hamming_distance}",
+            f"editdistance={inputs.processing.edit_distance}",
+            f"hammingdistance2={inputs.processing.hamming_distance2}",
+            f"qhdist2={inputs.processing.query_hamming_distance2}",
+            f"editdistance2={inputs.processing.edit_distance2}",
+            f"forbidn={inputs.processing.forbid_N}",
+            f"findbestmatch={inputs.processing.find_best_match}",
+            f"maskfullycovered={inputs.operations.mask_fully_covered}",
+            f"mink={inputs.operations.min_k}",
+            f"trimq={inputs.operations.trim_quality}",
+            f"qin={inputs.operations.quality_encoding_offset}",
+            f"trimpolya={inputs.operations.trim_poly_A}",
+            f"minlength={inputs.min_length}",
+            f"minlengthfraction={inputs.operations.min_length_fraction}",
+            f"minavgquality={inputs.operations.min_average_quality}",
+            f"maqb={inputs.operations.min_average_quality_bases}",
+            f"minbasequality={inputs.operations.min_base_quality}",
+            f"minconsecutivebases={inputs.operations.min_consecutive_bases}",
+            f"trimpad={inputs.operations.trim_pad}",
+            f"minoverlap={inputs.operations.min_overlap}",
+            f"mininsert={inputs.operations.min_insert}",
+            f"forcetrimleft={inputs.operations.force_trim_left}",
+            f"forcetrimright={inputs.operations.force_trim_right}",
+            f"forcetrimright2={inputs.operations.force_trim_right2}",
+            f"forcetrimmod={inputs.operations.force_trim_mod}",
+            f"restrictleft={inputs.operations.restrict_left}",
+            f"restrictright={inputs.operations.restrict_right}",
+            f"mingc={inputs.operations.min_GC}",
+            f"maxgc={inputs.operations.max_GC}",
+            f"maxns={inputs.operations.maxns}",
+            f"tossjunk={inputs.operations.toss_junk}",
+            f"chastityfilter={inputs.header_parsing.chastity_filter}",
+            f"barcodefilter={inputs.header_parsing.barcode_filter}",
+            f"xmin={inputs.header_parsing.x_min}",
+            f"ymin={inputs.header_parsing.y_min}",
+            f"xmax={inputs.header_parsing.x_max}",
+            f"ymax={inputs.header_parsing.y_max}",
+            f"entropy={inputs.complexity.entropy}",
+            f"entropywindow={inputs.complexity.entropy_window}",
+            f"entropyk={inputs.complexity.entropy_k}",
+            f"minbasefrequency={inputs.complexity.min_base_frequency}",
+            f"entropymask={inputs.complexity.entropy_mask}",
+            f"-Xmx{int(0.85*(self.requirements.resources.memory/num_of_lanes))}m",
+        ]
+
+        if self.requirements.resources.cores >= num_of_lanes:
+            args.append(
+                f"threads={int(self.requirements.resources.cores//num_of_lanes)}"
```

Comment:
If there are more lanes than cores this would be 0. Raising an informative error in such case should suffice.